parquet-common/src/main/java/org/apache/parquet/bytes/MultiBufferInputStream.java

```diff
@@ -238,8 +257,31 @@ public int read(byte[] bytes, int off, int len) {
   }
 
   @Override
-  public int read(byte[] bytes) {
-    return read(bytes, 0, bytes.length);
+  public void readFully(byte[] bytes, int off, int len) throws IOException {
+    if (len <= 0) {
+      if (len < 0) {
+        throw new IndexOutOfBoundsException("Read length must be greater than 0: " + len);
+      }
+      
+      return;
+    }
+
+    if (current == null || len > length) {
+      throw new EOFException();
+    }
+
+    int bytesRead = 0;
+    while (bytesRead < len) {
```

Comment:
This seems duplicate with above line 244.

Reply:
There are two key differences that make it hard to combine them without hurting performance for one, the other, or both, and they're both performance critical.

Reply:
Duplicating code is hard to maintain.

Reply:
Well, my objective here is to maximize performance. So we have to decide between maintainability and performance. Let's deliberate over this a bit more, and I'll do what you think is best.

Reply:
> Well, my objective here is to maximize performance. So we have to decide between maintainability and performance. Let's deliberate over this a bit more, and I'll do what you think is best.

I deeply understand this is a difficult trade-off. Do you have any evidence on the performance penalty if we wrap `read` and `readFully` methods to share some common logic? If the penalty is acceptable, we should definitely go for maintainability.